module1/homework/grayscaleImages/compression.hpp

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <array>
+#include <vector>
+
+constexpr size_t width = 32;
+constexpr size_t height = 32;
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>&);
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>&);
```

Comment:
You can simplify like this 
```suggestion
using Image = const std::array<std::array<uint8_t, width>, height>&;
using compressedImage = const std::vector<std::pair<uint8_t, uint8_t>>&;

std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(Image);
std::array<std::array<uint8_t, width>, height> decompressGrayscale(compressedImage );
```

Reply:
Why not like this?
```suggestion
using Image = std::array<std::array<uint8_t, width>, height>;
using compressedImage = std::vector<std::pair<uint8_t, uint8_t>>;

compressedImage compressGrayscale(const Image&);
Image decompressGrayscale(const compressedImage&);
```

Reply:
You are right :) it's better :)

Reply:
@Vicx95 Please never use sth like `using Image = const std::array<std::array<uint8_t, width>, height>&;` because this could make a lot of problem :)